idlparser/src/ffi/types/visitor.rs

```diff
@@ -0,0 +1,295 @@
+    raw_visitor::accept_func_param(
+        unsafe { func_param.raw_func_param.as_ref() }.unwrap(),
+        &mut visitor,
+    );
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_type(r#type: *const Type, visitor: *mut Visitor) {
+    let r#type = unsafe { r#type.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_type(unsafe { r#type.raw_type.as_ref() }.unwrap(), &mut visitor);
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_type_decl(type_decl: *const TypeDecl, visitor: *mut Visitor) {
+    let type_decl = unsafe { type_decl.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_type_decl(type_decl, &mut visitor);
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_struct_def(struct_def: *const StructDef, visitor: *mut Visitor) {
+    let struct_def = unsafe { struct_def.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_struct_def(struct_def, &mut visitor);
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_struct_field(
+    struct_field: *const StructField,
+    visitor: *mut Visitor,
+) {
+    let struct_field = unsafe { struct_field.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_struct_field(
+        unsafe { struct_field.raw_struct_field.as_ref() }.unwrap(),
+        &mut visitor,
+    );
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_enum_def(enum_def: *const EnumDef, visitor: *mut Visitor) {
+    let enum_def = unsafe { enum_def.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_enum_def(enum_def, &mut visitor);
+}
+
+/// # Safety
+///
+/// See documentation for [`const_ptr::as_ref`] and [`mut_ptr::as_mut`]
+#[no_mangle]
+pub unsafe extern "C" fn accept_enum_variant(
+    enum_variant: *const EnumVariant,
+    visitor: *mut Visitor,
+) {
+    let enum_variant = unsafe { enum_variant.as_ref() }.unwrap();
+    let visitor = unsafe { visitor.as_mut() }.unwrap();
+    let mut visitor = VisitorWrapper(visitor);
+    raw_visitor::accept_enum_variant(
+        unsafe { enum_variant.raw_enum_variant.as_ref() }.unwrap(),
+        &mut visitor,
+    );
+}
+
+macro_rules! fn_ptr_addr {
```

Comment:
May we replace this macro with a function?

Reply:
Tried, but unfortunatelly could not achieve it